drakcore/drakcore/frontend/src/AnalysisLogs.js

```diff
@@ -0,0 +1,352 @@
+    for (let idx = startChunkIndex; idx <= endChunkIndex; idx++) {
+      const chunk = index.markers[idx];
+      const isLastChunk = idx === index.markers.length - 1;
+      const nextChunk = index.markers[idx + 1];
+      const endOffset = isLastChunk ? null : nextChunk.offset;
+
+      const promise = queryData(chunk.offset, endOffset).then((chunk) => {
+        return [idx, chunk];
+      });
+
+      promises.push(promise);
+    }
+
+    let updatedPromise = new Promise((resolve, _) => {
+      // Register resolver
+      pending.current = resolve;
+    });
+
+    // When all chunks are loaded, update state
+    Promise.all(promises).then((results) => {
+      const newChunks = {};
+      for (const [idx, chunk] of results) {
+        newChunks[idx] = chunk;
+      }
+      setChunks({ ...chunks, ...newChunks });
+    });
+
+    return updatedPromise;
+  };
+
+  const loader = function ({ width, height }) {
+    return (
+      <InfiniteLoader
+        isItemLoaded={isItemLoaded}
+        loadMoreItems={loadMoreItems}
+        itemCount={lines}
+        minimumBatchSize={2000}
+      >
+        {({ onItemsRendered, ref }) => (
+          <List
+            height={height}
+            width={width}
+            itemCount={lines}
+            itemSize={30}
+            onItemsRendered={onItemsRendered}
+            ref={ref}
+          >
+            {(props) => {
+              const lineIndex = props.index;
+
+              const chunkIndex = findChunkIndex(lineIndex);
+              const chunk = chunks[chunkIndex];
+              if (chunk === undefined) {
+                return children({
+                  entry: undefined,
+                  index: lineIndex,
+                  style: props.style,
+                });
+              }
+              const lineOffset = index ? index.markers[chunkIndex].line : 0;
+              const entry = chunk[lineIndex - lineOffset];
+              return children({
+                entry,
+                index: lineIndex,
+                style: props.style,
+              });
+            }}
+          </List>
+        )}
+      </InfiniteLoader>
+    );
+  };
+  return <AutoSizer>{loader}</AutoSizer>;
+}
+
+const SERVICE_LOGS = ["drakrun.log", "drak-postprocess.log"];
+
+const DrakvufRow = ({ entry, style }) => {
+  let content;
+  let displayed = { ...entry };
+  if (entry === undefined) {
+    content = <>Loading...</>;
+  } else {
+    content = (
+      <code style={{ whiteSpace: "nowrap" }}>{JSON.stringify(displayed)}</code>
+    );
+  }
+
+  return (
+    <div style={style} className="d-flex flex-row align-content-stretch">
+      <div className="p-1">{content}</div>
+    </div>
+  );
+};
+
+function formatTimestamp(timestamp) {
+  const newDate = new Date();
+  newDate.setTime(timestamp * 1000);
+  const dateString = newDate.toUTCString();
+  return dateString;
+}
+
+const ServiceRow = ({ entry, index, style }) => {
+  if (entry === undefined) {
```

Comment:
Pretty sure correct way to check this is
```suggestion
  if (typeof entry === "undefined") {
```